### DISCUSSION TITLE
addHook

### DISCUSSION THREAD

**Discussion by rafpor75** [mia-platform/community#522] [Q&A]
### Description

Hi,
if I update @mia-platform/custom-plugin-lib from 5.0.0 to 7.0.0, I can't use with DecoratedFastify the addHook function, it doesn't exist anymore, is there an alternative? because it is present too  into last version of fastify original library

**Comment by davidebianchi** [comment]
Hi @rafpor75! Can you share an example of the implementation? I tried to add a `addHook` function and it is correctly invoked.
It could be an issue with typescript types, or are you using plain js?

**Reply by rafpor75** [reply]
Yes, it's type script:

export type UndecoratedMainService = DecoratedFastify<EnvVariables> & {
  customMetrics: Metrics
}

export type MainService = UndecoratedMainService & {
  crudClient: CrudClient
  timerServiceClient: TimerServiceClient
  fileServiceClient: FileServiceClient
  eventEmitter: EventEmitter
}

and 

export default async function myPlugin(service: MainService) {
//it can't use while with 5.0.0 yes
service.addHook(etc, etc){

}
}

**Reply by davidebianchi** [reply]
I tried with the version 7 of custom-plugin-lib with your example, and it works correctly also in ts. Can you give some more information?

**Comment by rafpor75** [comment]
it isn't an issue
